File: code-examples/examples/example-4/src/test/java/be/swsb/typesafety/example4/external/legacybookingsystem/LegacyMapperTest.java
package be.swsb.typesafety.example4.external.legacybookingsystem;

import be.swsb.typesafety.example4.domain.booking.Booking;
import be.swsb.typesafety.example4.domain.booking.BookingCode;
import be.swsb.typesafety.example4.domain.booking.BookingTestBuilder;
import be.swsb.typesafety.example4.external.legacybookingsystem.acl.types.LegacyBookingType;
import be.swsb.typesafety.test.UnitTest;
import org.junit.jupiter.api.Test;

import java.util.Optional;

import static be.swsb.typesafety.example4.domain.booking.BookingCodeForTests.randomBookingCode;
import static be.swsb.typesafety.example4.domain.booking.BookingTestBuilder.defaultBooking;
import static org.assertj.core.api.Assertions.assertThat;

class LegacyMapperTest extends UnitTest {

    private LegacyMapper mapper = new LegacyMapper();

    @Test
    void toXml() {
        final Optional<BookingCode> bookingCode = randomBookingCode();
        final Booking booking = defaultBooking()
                .withBookingCode(bookingCode)
                .build();

        final LegacyBookingType legacyBookingType = mapper.toXml(booking);

        assertThat(legacyBookingType.getBookingCode()).isEqualTo(bookingCode.get().asString());
    }

    @Test
    void toXml_BookingCodeIsEmpty_NoCodeOnXml() {
        final Booking booking = defaultBooking()
                .withBookingCode(Optional.empty())
                .build();

        final LegacyBookingType legacyBookingType = mapper.toXml(booking);

        assertThat(legacyBookingType.getBookingCode()).isNull();
    }
}

File: code-examples/examples/example-5/src/test/java/be/swsb/typesafety/example5/external/legacybookingsystem/acl/mapper/LegacyMapperTest.kt
package be.swsb.typesafety.example5.external.legacybookingsystem.acl.mapper

import be.swsb.typesafety.example5.domain.booking.Booking
import be.swsb.typesafety.example5.domain.booking.BookingCode
import be.swsb.typesafety.example5.domain.booking.randomBookingCode
import be.swsb.typesafety.example5.external.legacybookingsystem.acl.mapper.LegacyMapper
import be.swsb.typesafety.test.UnitTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class LegacyMapperTest : UnitTest() {

    private val mapper = LegacyMapper()

    @Test
    fun toXml() {
        val bookingCode = BookingCode.randomBookingCode()
        val booking = Booking.booking(bookingCode)

        val legacyBookingType = mapper.toXml(booking)

        assertThat(legacyBookingType.bookingCode).isEqualTo(bookingCode?.asString())
    }

    @Test
    fun toXml_BookingCodeIsEmpty_NoCodeOnXml() {
        val booking = Booking.booking(null)

        val legacyBookingType = mapper.toXml(booking)

        assertThat(legacyBookingType.bookingCode).isNull()
    }
}